### PR TITLE
DNS: add Nebulo as worth mentioning & warn about DoH metadata & sort worth mentioning/additional information into sublists

### PR DESCRIPTION
**Pull request by Mikaela**
## Description

Resolves: #1187

#### Check List 

- [x] I have read and understand [the contributing guidelines](https://github.com/privacytoolsIO/privacytools.io/blob/master/.github/CONTRIBUTING.md).

- [x] I have [listed the source code](https://github.com/privacytoolsIO/privacytools.io/blob/master/source_code.md).

- [x] This project is [free/libre software](https://www.wikipedia.org/wiki/Free_software).

- [x] This project has an [associated discussion](https://github.com/privacytoolsIO/privacytools.io/issues).

* Netlify preview for the mainly edited page: https://deploy-preview-1200--privacytools-io.netlify.com/providers/dns/#icanndns

* Code Repository (if applicable): https://git.frostnerd.com/PublicAndroidApps/smokescreen/

### CODE REVIEW

File: _includes/sections/dns.html [hunk 1]
```diff
@@ -281,7 +281,7 @@ <h4>Terms</h4>
 
   <ul>
     <li>DNS-over-TLS (DoT) - A security protocol for encrypted DNS on a dedicated port 853. Some providers support port 443 which generally works everywhere while port 853 is often blocked by restrictive firewalls.</li>
-    <li>DNS-over-HTTPS (DoH) - Similar to DoT, but uses HTTPS instead, being indistinguishable from "normal" HTTPS traffic on port 443.</li>
+    <li>DNS-over-HTTPS (DoH) - Similar to DoT, but uses HTTPS instead, being indistinguishable from "normal" HTTPS traffic on port 443. <span class="badge badge-warning" data-toggle="tooltip" data-original-title="DoH may leak metadata such as user-agent (which may include system information) to the DNS server"><a href="https://github.com/privacytoolsIO/privacytools.io/issues/1187#issuecomment-523313881"><i class="fas fa-exclamation-triangle"></i></a></span></li>
```
Comment (Mikaela) [inline _includes/sections/dns.html]: TODO: find a more authoritative source than someone on our issue tracker? :)

Comment (Mikaela) [reply _includes/sections/dns.html]: Show me a more authoritative source than the RFC!

File: _includes/sections/dns.html [hunk 2]
```diff
@@ -281,7 +281,7 @@ <h4>Terms</h4>
 
   <ul>
     <li>DNS-over-TLS (DoT) - A security protocol for encrypted DNS on a dedicated port 853. Some providers support port 443 which generally works everywhere while port 853 is often blocked by restrictive firewalls.</li>
-    <li>DNS-over-HTTPS (DoH) - Similar to DoT, but uses HTTPS instead, being indistinguishable from "normal" HTTPS traffic on port 443.</li>
+    <li>DNS-over-HTTPS (DoH) - Similar to DoT, but uses HTTPS instead, being indistinguishable from "normal" HTTPS traffic on port 443. <span class="badge badge-warning" data-toggle="tooltip" data-original-title="DoH contains metadata such as user-agent (which may include system information) to the DNS server"><a href="https://tools.ietf.org/html/rfc8484#section-8.2"><i class="fas fa-exclamation-triangle"></i></a></span></li>
```
Comment (Mikaela) [inline _includes/sections/dns.html]: English: Do things contain or include metadata?

Comment (nitrohorse) [reply _includes/sections/dns.html]: I think "contains" is fine. I think we could also make this a tad more readable by:
> DoH contains metadata such as user-agent (which may include system information) **that is sent** to the DNS server. (also add a period)

File: _includes/sections/dns.html [hunk 3]
```diff
@@ -297,5 +297,6 @@ <h3>Worth Mentioning and Additional Information</h3>
     <li><a href="https://namecoin.info/">Namecoin</a> - A decentralized DNS open-source information registration and transfer system based on the Bitcoin cryptocurrency.</li>
     <li><a href="https://www.isc.org/blogs/qname-minimization-and-privacy/">QNAME Minimization and Your Privacy</a> by the Internet Systems Consortium (ISC)</li>
     <li><a href="https://www.isc.org/dnssec/">DNSSEC and BIND 9</a> by the ISC</li>
+    <li><a href="https://git.frostnerd.com/PublicAndroidApps/smokescreen/blob/master/README.md">Nebulo</a> - An open-source application for Android supporting DoH and DoT. It also supports caching DNS responses and locally logging DNS queries.</li>
```
Comment (nitrohorse) [inline _includes/sections/dns.html]: Could we move this directly above or below DNSCloak just to group similar apps together?

File: _includes/sections/dns.html (hunk 2 above)
Comment (Mikaela) [reply _includes/sections/dns.html]: Sure, will do

File: _includes/sections/dns.html (hunk 3 above)
Comment (Mikaela) [reply _includes/sections/dns.html]: How about copying the XMPP syntax from https://www.privacytools.io/software/real-time-communication/#im to do it?

I didn't see the logic before, so I just put it in order of addition. I will try to commit it so we will see how it looks like.

Comment (Mikaela) [reply _includes/sections/dns.html]: I think it looks better, but now I have to sort everything else too, not that it's a bad thing:

image

I fear we may be causing merge conflicts for each other depending on which PRs are merged, but if that happens, we can resolve them.

**Comment by Mikaela** [comment]
So currently it looks like this:

image

File: _includes/sections/dns.html [hunk 4]
```diff
@@ -281,21 +281,42 @@ <h4>Terms</h4>
...
-      <li><a href="https://pi-hole.net/">Pi-hole</a> - A network-wide DNS server mainly for the Raspberry Pi. Blocks ads, tracking, and malicious domains for all devices on your network.</li>
-      <li><a href="https://gitlab.com/quidsup/notrack">NoTrack</a> - A network-wide DNS server like Pi-hole for blocking ads, tracking, and malicious domains.</li>
-    <li><a href="https://dnsprivacy.org/wiki/display/DP/DNS+Privacy+Daemon+-+Stubby">Stubby</a> - An open-source application for Linux, macOS, and Windows that acts as a local DNS Privacy stub resolver using DoT.</li>
-    <li><a href="https://namecoin.info/">Namecoin</a> - A decentralized DNS open-source information registration and transfer system based on the Bitcoin cryptocurrency.</li>
-    <li><a href="https://www.isc.org/blogs/qname-minimization-and-privacy/">QNAME Minimization and Your Privacy</a> by the Internet Systems Consortium (ISC)</li>
-    <li><a href="https://www.isc.org/dnssec/">DNSSEC and BIND 9</a> by the ISC</li>
+    <li><strong>Desktop software with support for encrypted DNS:</strong></li>
```
Comment (Mikaela) [inline _includes/sections/dns.html]: I understand that we don't have a mobile-first attitude, so desktop software is what a visitor is most likely looking for and for now it's not so long, so I think it makes the most sense being on top.

File: _includes/sections/dns.html [hunk 5]
```diff
@@ -281,21 +281,42 @@ <h4>Terms</h4>
...
+      <ul>
+        <li><em>Firefox</em> comes with built-in DoH support with Cloudflare set as the default resolver, but can be configured to use any DoH resolver. <span class="badge badge-warning" data-toggle="tooltip" data-placement="bottom" data-original-title='"Cloudflare has agreed to collect only a limited amount of data about the DNS requests that are sent to the Cloudflare Resolver for Firefox via the Firefox browser."'><a href="https://developers.cloudflare.com/1.1.1.1/commitment-to-privacy/privacy-policy/firefox/"><i class="fas fa-exclamation-triangle"></i></a></span> Currently Mozilla is <a href="https://blog.mozilla.org/futurereleases/2019/07/31/dns-over-https-doh-update-detecting-managed-networks-and-user-choice/">conducting studies</a> before enabling DoH by default for all US-based Firefox users.</li>
+      </ul>
+
+    <li><strong>Mobile software with support for encrypted DNS:</strong></li>
```
Comment (Mikaela) [inline _includes/sections/dns.html]: I think it's the second most likely thing someone is looking for

File: _includes/sections/dns.html [hunk 6]
```diff
@@ -281,21 +281,42 @@ <h4>Terms</h4>
...
+      <ul>
+        <li><em>Android 9</em> comes with a DoT client by <a href="https://support.google.com/android/answer/9089903">default</a>. <span class="badge badge-warning" data-toggle="tooltip" data-original-title="...but with some caveats"><a href="https://www.quad9.net/private-dns-quad9-android9/"><i class="fas fa-exclamation-triangle"></i></a></span></li>
+        <li><em><a href="https://apps.apple.com/app/id1452162351">DNSCloak</a></em> - An <a href="https://github.com/s-s/dnscloak">open-source</a> DNSCrypt and DoH client for iOS by <td><a data-toggle="tooltip" data-placement="bottom" data-original-title='"A charitable non-profit host organization for international Free Software projects."' href="https://techcultivation.org/">the Center for the Cultivation of Technology gemeinnuetzige GmbH</a>.</li>
+        <li><em><a href="https://git.frostnerd.com/PublicAndroidApps/smokescreen/blob/master/README.md">Nebulo</a></em> - An open-source application for Android supporting DoH and DoT. It also supports caching DNS responses and locally logging DNS queries.</li>
+      </ul>
+
+    <li><strong>Local DNS servers:</strong></li>
```
Comment (Mikaela) [inline _includes/sections/dns.html]: Anyone can run them, so I think it's a good third thing to list

File: _includes/sections/dns.html [hunk 7]
```diff
@@ -281,21 +281,42 @@ <h4>Terms</h4>
...
+      <ul>
+        <li><em><a href="https://dnsprivacy.org/wiki/display/DP/DNS+Privacy+Daemon+-+Stubby">Stubby</a></em> - An open-source application for Linux, macOS, and Windows that acts as a local DNS Privacy stub resolver using DoT.</li>
+        <li><em><a href="https://namecoin.info/">Namecoin</a></em> - A decentralized DNS open-source information registration and transfer system based on the Bitcoin cryptocurrency.</li>
+      </ul>
+
+  <li><strong>Network wide DNS servers:</strong></li>
```
Comment (Mikaela) [inline _includes/sections/dns.html]: I think it's the most technical thing we list and requires the most, so it's good to be on the bottom of recommendations.

File: _includes/sections/dns.html [hunk 8]
```diff
@@ -281,21 +281,42 @@ <h4>Terms</h4>
...
+    <ul>
+      <li><em><a href="https://pi-hole.net/">Pi-hole</a></em> - A network-wide DNS server mainly for the Raspberry Pi. Blocks ads, tracking, and malicious domains for all devices on your network.</li>
+      <li><em><a href="https://gitlab.com/quidsup/notrack">NoTrack</a></em> - A network-wide DNS server like Pi-hole for blocking ads, tracking, and malicious domains.</li>
+    </ul>
+
+    <li><strong>Further reading:</strong></li>
```
Comment (Mikaela) [inline _includes/sections/dns.html]: I first thought about putting it on top, but as the heading says "Worth mentioning and additional information" I thought it should be on bottom and I cannot think of anything better here. Isn't "further reading" generally something that one doesn't have to read, but which can be of interest to someone who is especially interested in the topic?

File: source_code.md [hunk 9]
```diff
@@ -285,6 +285,8 @@ OpenNic: https://github.com/opennic/ (mostly)
 
 Webpage: https://github.com/opennic/opennic-web
 
+Nebulo: https://git.frostnerd.com/PublicAndroidApps/smokescreen/
```
Comment (Mikaela) [inline source_code.md]: I didn't start touching this file as there hasn't been feedback (and I forgot it), but should I start sorting the recommendations there? I hope not.

File: _includes/sections/dns.html (hunk 4 above)
Comment (nitrohorse) [reply _includes/sections/dns.html]: Yeah, makes sense to me :+1:

File: _includes/sections/dns.html (hunk 5 above)
Comment (nitrohorse) [reply _includes/sections/dns.html]: I wonder if these titles are as clear and a bit less wordy?
- `Encrypted DNS clients for desktop:`
- `Encrypted DNS clients for mobile:`

File: _includes/sections/dns.html (hunk 8 above)
Comment (nitrohorse) [reply _includes/sections/dns.html]: Yeah, makes sense to me!

File: source_code.md (hunk 9 above)
Comment (nitrohorse) [reply source_code.md]: Ideally I think each section would have the same order as what's on the section but I think as long as the source code is listed in the right section that's fine.

File: _includes/sections/dns.html (hunk 4 above)
Comment (nitrohorse) [inline _includes/sections/dns.html]: I think just to keep the HTML tidy, we should move the `</li>` tags to the bottom of their corresponding `</ul>` tag, wrapping the list. Ex:
```html
  <ul>
    <li><strong>Desktop software with support for encrypted DNS:</strong>
      <ul>
        <li><em>Firefox</em> comes with built-in DoH support with Cloudflare set as the default resolver, but can be configured to use any DoH resolver. <span class="badge badge-warning" data-toggle="tooltip" data-placement="bottom" data-original-title='"Cloudflare has agreed to collect only a limited amount of data about the DNS requests that are sent to the Cloudflare Resolver for Firefox via the Firefox browser."'><a href="https://developers.cloudflare.com/1.1.1.1/commitment-to-privacy/privacy-policy/firefox/"><i class="fas fa-exclamation-triangle"></i></a></span> Currently Mozilla is <a href="https://blog.mozilla.org/futurereleases/2019/07/31/dns-over-https-doh-update-detecting-managed-networks-and-user-choice/">conducting studies</a> before enabling DoH by default for all US-based Firefox users.</li>
      </ul>
    </li>
    <li><strong>Mobile software with support for encrypted DNS:</strong>
      <ul>
        <li><em>Android 9</em> comes with a DoT client by <a href="https://support.google.com/android/answer/9089903">default</a>. <span class="badge badge-warning" data-toggle="tooltip" data-original-title="...but with some caveats"><a href="https://www.quad9.net/private-dns-quad9-android9/"><i class="fas fa-exclamation-triangle"></i></a></span></li>
        <li><em><a href="https://apps.apple.com/app/id1452162351">DNSCloak</a></em> - An <a href="https://github.com/s-s/dnscloak">open-source</a> DNSCrypt and DoH client for iOS by <td><a data-toggle="tooltip" data-placement="bottom" data-original-title='"A charitable non-profit host organization for international Free Software projects."' href="https://techcultivation.org/">the Center for the Cultivation of Technology gemeinnuetzige GmbH</a>.</li>
        <li><em><a href="https://git.frostnerd.com/PublicAndroidApps/smokescreen/blob/master/README.md">Nebulo</a></em> - An open-source application for Android supporting DoH and DoT. It also supports caching DNS responses and locally logging DNS queries.</li>
      </ul>
    </li>
    <li><strong>Local DNS servers:</strong>
      <ul>
        <li><em><a href="https://dnsprivacy.org/wiki/display/DP/DNS+Privacy+Daemon+-+Stubby">Stubby</a></em> - An open-source application for Linux, macOS, and Windows that acts as a local DNS Privacy stub resolver using DoT.</li>
        <li><em><a href="https://namecoin.info/">Namecoin</a></em> - A decentralized DNS open-source information registration and transfer system based on the Bitcoin cryptocurrency.</li>
      </ul>
    </li>
    <li><strong>Network wide DNS servers:</strong>
      <ul>
        <li><em><a href="https://pi-hole.net/">Pi-hole</a></em> - A network-wide DNS server mainly for the Raspberry Pi. Blocks ads, tracking, and malicious domains for all devices on your network.</li>
        <li><em><a href="https://gitlab.com/quidsup/notrack">NoTrack</a></em> - A network-wide DNS server like Pi-hole for blocking ads, tracking, and malicious domains.</li>
      </ul>
    </li>
    <li><strong>Further reading:</strong>
      <ul>
        <li><a href="https://www.isc.org/blogs/qname-minimization-and-privacy/">QNAME Minimization and Your Privacy</a> by the Internet Systems Consortium (ISC)</li>
        <li><a href="https://www.isc.org/dnssec/">DNSSEC and BIND 9</a> by the ISC</li>
      </ul>
    </li>
  </ul>
```

**Comment by nitrohorse** [comment]
Some further suggestions but LGTM otherwise; glad to see this further organized! :smile:

File: source_code.md (hunk 9 above)
Comment (Mikaela) [reply source_code.md]: I tried to change the order to that

**Pull request merged**